guides/09_other-tutorials/wrapping-layouts.md

```diff
@@ -0,0 +1,179 @@
+        # other assign functions 
+
+        if render:
+            self.render()
+```
+
+So our implementation looks like this:
+
+```python
+# self.main_layout -> Row()
+with self.main_layout:
+    left_textbox.render()
+    right_textbox.render()
+```
+
+So with calling the components' render functions under the ```with``` syntax, we are simulating the default implementation, actually.
+
+If we understand the below ```with``` syntax, let's move on the upper ```with``` syntax. For this, let's expand our example with ```Tab``` component:
+
+```python
+with Tab():
+    with Row():
+        first_textbox = Textbox(value="first_textbox")
+        second_textbox = Textbox(value="second_textbox")
+```
+
+Pay attention to the Row and Tab components this time. We have created the Textbox variables above and added them to Row with ```with``` syntax. Now we need to add the Row component to the Tab component. You can see that Row component is created with default parameters, so its render parameter is true, that's why the render function is going to be executed under the Tab component's ```with``` syntax.
+
+To mimic this implementation, we need to call the ```render``` function of the ```main_layout``` variable after the ```with``` syntax of the ```main_layout``` variable.
+
+So the implementation looks like this:
+
+```python
+with tab_main_layout:
+    with row_main_layout:
+        first_textbox.render()
+        second_textbox.render()
+
+    row_main_layout.render()
+
+tab_main_layout.render()
+```
+
+The default implementation and our implementation are the same, but we are using the render function ourselves. So it requires a little work.
+
+Now, let's look at the ```attach_event``` function.
+
+2. Attach Event Function
+
+    The function is left as not implemented because it is specific to the class, so each class has to implement its `attach_event` function.
+
+```python
+    # other LayoutBase implementations
+
+    def attach_event(self, block_dict: Dict[str, Block]) -> None:
+        raise NotImplementedError
+```
+
+You can see what is the ```block_dict``` variable in the ```Application``` class's ```attach_event``` function.
+
+### Application Class
+
+1. Render Function
+
+```python
+    # other Application implementations
+
+    def _render(self):
+        with self.app:
+            for child in self.children:
+                child.render()
+
+        self.app.render()
+```
+
+From the explanation of the ```LayoutBase``` class's ```render``` function, we can understand the ```child.render``` part.
+
+So let's look at the below part, why we are calling the ```app``` variable's ```render``` function? We are calling this render function because if we look at the implementation of this function in the ```gradio.blocks.Blocks``` class, we can see that it is adding the components and event functions into the root component. With another saying, it is creating and structuring the gradio application. So it is important to call this function.
```

Comment:
```suggestion
So let's look at the bottom part, why are we calling the ```app``` variable's ```render``` function? It's important to call this function because if we look at the implementation in the ```gradio.blocks.Blocks``` class, we can see that it is adding the components and event functions into the root component. To put it another way, it is creating and structuring the gradio application. 
```